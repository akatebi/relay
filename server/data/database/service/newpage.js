
export const postBody = ({
  org,
  type,
  layout,
  title,
  documentKind,
}) => {
  const object = {
    organization: {
      isFamily: true,
      id: org,
      entityType: 'Organization',
    },
    documentType: {
      isFamily: true,
      id: type,
      entityType: 'DocumentType',
    },
    layoutRevision:
    {
      isFamily: false,
      id: layout,
      entityType: 'Layout',
    },
    documentKind,
    title,
  };
  if (layout === 'null') {
    delete object.layoutRevision;
  }
  return object;
};

export const query = (
  propertyName,
  value,
  isFamily,
) => ({
  baseQueryType: null,
  isFamilyType: false,
  cultureCode: '',
  filters: [
    {
      $type: 'Models.QueryCriteria.Filter, Models',
      filterType: 'Eq',
      propertyName,
      value,
      isFamily,
    },
  ],
  sortOrder: [],
  skip: 0,
  top: 0,
  inlineCount: 0,
  select: null,
  isFamilyTypeResult: false,
});
