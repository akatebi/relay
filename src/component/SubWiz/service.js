
export const selectPropsMap = list =>
  list.map(item => ({
    value: item.id ? item.id.slice(0, 36) : item.value,
    label: item.label,
    cultureCode: item.cultureCode,
    entityType: item.entityType,
    familyId: item.familyId,
    id: item.id,
    isFamily: item.isFamily,
    operation: item.operation,
    reportingKey: item.reportingKey,
  }));

// used by AppSegments
const getSegmentList = (approvalCycle) => {
  const segments = approvalCycle.approvalSegmentConfigs;

  const result = segments.length ?
    segments.map(segment => ({
      id: segment.id,
      segmentType: segment.segmentType,
      segmentLabel: segment.label,
      waitForEverybody: segment.waitForEverybody,
      placeholder: 'select approvers...',
      approverSelectionMode: segment.selectAllUsers ? 'Select Users' : 'All Matching Users',
      approverSource: segment.approverSource,
      approvers: {
        eligible: [],
        selected: [],
      },
    })) :
    ({
      id: '',
      segmentType: 'none',
      segmentLabel: 'none',
      waitForEverybody: false,
      placeholder: 'none',
      approverSelectionMode: 'none',
      approverSource: 'none',
      approvers: {
        eligible: [{ value: 'none', label: 'none' }],
        selected: [],
      },
    });
  // console.log('>>> getSegmentList result', window.pretty(result));
  return result;
};

// used by AppSegments
export const approvalCycleMap = approvalCycle =>
  ({
    id: approvalCycle.id,
    entityType: approvalCycle.entityType,
    label: approvalCycle.label,
    processType: approvalCycle.processType,
    waitForEverybody: approvalCycle.waitForEverybody,
    segmentList: getSegmentList(approvalCycle),
  });

const getAssigneesNew = (selectedApprovers) => {
  // console.log('selectedApprovers', window.pretty(selectedApprovers));
  const list = selectedApprovers.map(approver =>
    approver.reduce((acc, { segmentId, id }) => {
      acc.segmentId = segmentId;
      const obj = { id: id.slice(0, 36), isFamily: true, entityType: 'ApplicationRole' };
      acc.assignees.push(obj);
      return acc;
    }, { assignees: [] }));
  // console.log('list', window.pretty(list));
  return list;
};

const getAppCycleConfig = selectedCycleId => ({
  isFamily: true,
  id: selectedCycleId,
  entityType: 'ApprovalCycleConfig',
});

// used by AppSegments.onSubmit
// ==============================
export const mapSubmitData = ({
  lifeCycle,
  processType,
  selectedCycleId,
  selectedApprovers,
}) => {
  let toState = null;
  if (processType === 'approval') {
    toState = lifeCycle.lifeCycleStateContents
      .find(stateContent => stateContent.state === 'Approved');
    delete toState.__id;
  }
  return {
    toState,
    directApproval: false,
    isFamily: true,
    segmentAssignees: getAssigneesNew(selectedApprovers),
    approvalCycleConfig: getAppCycleConfig(selectedCycleId),
  };
};
