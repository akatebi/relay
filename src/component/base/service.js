
// const debug = require('debug')('app:root:service');

export const getGUID = pathname =>
  pathname.split('/').find(item =>
    /^[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$/.test(item));

export const allParams = ({ location, match, params = {}, name = '' }) => {
  const path = `${match.path}/${name}`.slice(1);
  const pathname = location.pathname.slice(1);
  const pathList = path.split('/');
  const pathnameList = pathname.split('/');
  // debug(name, 'pathList', pathList);
  // debug(name, 'pathnameList', pathnameList);
  const newParams = pathList.reduce((acc, prop, index) => {
    const value = pathnameList[index];
    // debug(name, 'prop', prop);
    if (prop[0] === ':') {
      acc[prop.slice(1)] = value;
    }
    return acc;
  }, {});
  // debug(name, 'newParams', window.pretty(newParams));
  const paramsAll = { ...params, ...newParams };
  // debug(name, 'params keys', Object.keys(paramsAll));
  return paramsAll;
};
