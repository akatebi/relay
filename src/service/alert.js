import React from 'react';
import qs from 'qs';
import Alert from 'react-s-alert';
import moment from 'moment';

// const debug = require('debug')('app:service:alert');

const options = {
  position: 'top-right',
  effect: 'slide',
  beep: false,
  timeout: 3000,
  offset: 0,
};

const timeString = (timeStr) => {
  const fromNow = `${moment(timeStr).fromNow()}`;
  const dateFmt = `${moment(timeStr).format('ll')} ${moment(timeStr).format('LT')}`;
  const dateOut = `${fromNow} on ${dateFmt}`;
  return dateOut;
};

const getMsg = (msg, eventTime) =>
  (eventTime ? `${msg} - ${timeString(eventTime)}` : msg);

export const successAlert = (msg, eventTime) => {
  setTimeout(() => {
    const message = getMsg(msg, eventTime);
    Alert.success(message, options);
  }, 1000);
};

export const infoAlert = (msg, eventTime) => {
  let message;
  if (!eventTime || eventTime === '0001-01-01T00:00:00Z') {
    message = msg;
  } else {
    message = getMsg(msg, eventTime);
  }
  Alert.info(message, options);
};

export const errorAlert = (msg, eventTime) => {
  const message = getMsg(msg, eventTime);
  if (/^https/.test(message)) {
    const { id } = qs.parse(message.split('_queries?')[1]);
    Alert.error(<a target="foo" href={message}>{`BUG: ${id}`}</a>, { ...options, timeout: 10000 });
  } else {
    Alert.error(message, { ...options, timeout: 10000 });
  }
};
