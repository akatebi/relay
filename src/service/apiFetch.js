import deepmerge from 'deepmerge';
import { serverPath, getToken } from '../constant/app';

const debug = require('debug')('app:service:apiFetch');

function apiFetch(apiUrl, extraOptions = {}) {
  const url = `${serverPath}/${apiUrl}`;
  const Authorization = `Bearer ${getToken()}`;
  const options = deepmerge({
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
      Authorization,
    },
  }, extraOptions);
  debug(url);
  // debug(options);
  return fetch(url, options)
    .then((resp) => {
      if (!resp.ok) {
        const error = { error: `${resp.status}-${resp.statusText}` };
        throw error;
      }
      // debug('resp:', resp);
      return resp;
    })
    .catch((error) => {
      debug('API-FETCH Error:', error);
      throw error;
    });
}

export default apiFetch;

export const myFetch = (url, extraOptions = {}) => {
  const Authorization = `Bearer ${getToken()}`;
  const options = deepmerge({
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
      Authorization,
    },
  }, extraOptions);
  // debug(url);
  // debug(options);
  return fetch(url, options);
};
