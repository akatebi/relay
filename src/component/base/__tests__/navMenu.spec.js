import { getToken, cleanLog, writeFile, graphql, fixGUIDS } from '../../__tests__/service';
import query from './query/navMenu';

// const debug = require('debug')('app:server:data:navMenu');

const dirname = `${__dirname}/logs/navMenu`;
const writeNav = writeFile.bind(this, dirname);

beforeAll(getToken);
beforeAll(cleanLog.bind(this, dirname));

test('navMenu', async () => {
  const { token } = global;
  const result = await graphql(query, { token });
  expect(result.errors).toBeUndefined();
  await writeNav('result', result);
  const { userIdentity } = result.data.viewer;
  expect(userIdentity.token.length).toBeGreaterThan(500);
  delete userIdentity.token;
  expect(fixGUIDS(result.data)).toMatchSnapshot();
});
