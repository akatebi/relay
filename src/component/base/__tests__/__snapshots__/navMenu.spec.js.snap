// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navMenu 1`] = `
Object {
  "viewer": Object {
    "error": null,
    "userIdentity": Object {
      "organization": Object {
        "entityType": "Organization",
        "id": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx-entityVM",
        "isFamily": true,
        "label": "Enterprise Organization",
        "revId": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx",
      },
      "user": Object {
        "entityType": "User",
        "id": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx-entityVM",
        "isFamily": false,
        "label": "User0",
        "revId": "",
      },
    },
  },
}
`;
